client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -106,8 +108,11 @@ public boolean hasNext() {
       shuffleReadMetrics.incFetchWaitTime(fetchDuration);
       if (compressedData != null) {
         shuffleReadMetrics.incRemoteBytesRead(compressedData.limit() - compressedData.position());
+        if (uncompressedData != null && uncompressedData.isDirect()) {
+          ((DirectBuffer)uncompressedData).cleaner().clean();
```

Comment:
can it be compatible with jdk9 or higher?

Reply:
Have not try